### PR TITLE
Add validation

### PR DESCRIPTION
**Pull request by aljones15**
This PR has been tested with the tests in this project and in the wallet and issuer. It works in all cases to my knowledge

1. Wallet Registration works
2. Issuer manage works
3. Issuer onboarding works

Moving this from Draft to actual Pr after fixing the zcap validator.

### CODE REVIEW

**Comment by mattcollier** [comment]
@aljones15 great!  Regarding the `zcaps` being an array vs an object.  An older iteration of the `zcaps` data model had them as an array and in practical usage in the wallet and issuer it should be an object.  Where are you finding it still as an array?  If it's just in the test suite, then those tests just need to be updated.  I believe we *do* want the validator to ensure `zcaps` is in an object and we need to fix anything that breaks based on that.

**Comment by aljones15** [comment]
@mattcollier  so basically zcaps is just a validator ensuring zcaps is an object and that's it? or is there some way to ensure a property of an object conforms to a schema with out knowing the name? or do we know the names of the zcaps properties? 

WRT to the array it's the test data. I'll try to get this change in today.

**Comment by mattcollier** [comment]
We won't know the key names in the zcaps object, but each value will be a zcap that should eventually be able to be validated against some zcap schema TBD.  I'm not aware of any existing schema for a zcap.

**Comment by aljones15** [comment]
@mattcollier the unknown properties can be validated by passing a validator to `additionalProperties` https://json-schema.org/understanding-json-schema/reference/object.html

so trying that.

File: schemas/bedrock-profile-http.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    allowedAction: {
+      anyOf: [{
+        type: 'string'
+      }, {
+        type: 'array',
+        minItems: 1,
+        items: [{type: 'string'}]
+      }]
+    },
+    invoker: {
```
Comment (dlongley) [inline schemas/bedrock-profile-http.js:101]: zcaps can also have `delegator`; it looks like `invoker` does.

Comment (aljones15) [reply schemas/bedrock-profile-http.js:101]: I added `delegator` to the zcap. It is also missing `referenceId`, `parentCapability`, and `proof`. https://github.com/digitalbazaar/bedrock-profile-http/pull/10/commits/db5cc0508923ab42a904fe0de5931d4090682b52

**Comment by mattcollier** [comment]
@aljones15 this is looking good. Please rebase on master to pull in CI stuff.

Also, just want to ensure that all the latest changes were run through some wallet/issuer flow as you mentioned in the PR description.

File: schemas/bedrock-profile-http.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      type: 'object'
+    },
+    '@context': {
+      title: '@context',
+      anyOf: [{
+        type: 'string'
+      }, {
+        type: 'array',
+        minItems: 1,
+        items: {anyOf: [{type: 'string'}, {type: 'object'}]}
```
Comment (dlongley) [inline schemas/bedrock-profile-http.js:108]: We shouldn't allow objects in the context.

File: schemas/bedrock-profile-http.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+          type: {
+            title: 'Invocation Target Type',
+            type: 'string'
+          }
+        }
+      }]
+    },
+    parentCapability: {
+      title: 'Parent Capability',
+      anyOf: [{type: 'string'}, {type: 'object'}]
```
Comment (dlongley) [inline schemas/bedrock-profile-http.js:155]: Do we ever use objects here? If not, let's restrict this to just strings.

Comment (aljones15) [reply schemas/bedrock-profile-http.js:155]: That might cause issues with other projects. I think restricting it to just a string is to strong.

Comment (aljones15) [reply schemas/bedrock-profile-http.js]: Addressed here: https://github.com/digitalbazaar/bedrock-profile-http/pull/10/commits/77038a0960cdf291be06ec761367bea3217b5a11

File: schemas/bedrock-profile-http.js (hunk 2 above)
Comment (aljones15) [reply schemas/bedrock-profile-http.js]: Addressed here: https://github.com/digitalbazaar/bedrock-profile-http/pull/10/commits/77038a0960cdf291be06ec761367bea3217b5a11

**Comment by aljones15** [comment]
Wallet Tests:
  - Email based Registration passed
  - Create new Profile Permanent passed
  - Create new Profile Temporary passed

Issuer Tests:
  - Manage registration passed
  - Manage login passed
  - Add new instance passed
  - Instance Dashboard Add Users passed
  - Instance Dashboard Add Integration passed


I did run into one error during testing which involed a profile not being authorized to manage an profile edv or something like that. It was not validator related I believe I will try to replicate and file an issue.

@mattcollier tests listed on latest revisions.

File: schemas/bedrock-profile-http.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+const zcaps = {
+  title: 'zcaps',
+  type: 'object',
+  properties: {
+    zcaps: {
+      title: 'zcaps',
+      type: 'object',
+      // each property must be a zcap
+      additionalProperties: zcap
```
Comment (mattcollier) [inline schemas/bedrock-profile-http.js:174]: This is VERY cool, I didn't know it was a thing.

**Pull request merged**